src/lib/storage/table.cpp

```diff
@@ -17,32 +17,42 @@
 
 namespace opossum {
 
-Column::Column(std::string name, std::string type) : name(std::move(name)), type(std::move(type)) {}
+ColumnDefinition::ColumnDefinition(std::string name, std::string type) : name(std::move(name)), type(std::move(type)) {}
 
 Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size{target_chunk_size} {
   // create initial chunk
-  _append_new_chunk();
+  create_new_chunk();
 }
 
 void Table::add_column_definition(const std::string& name, const std::string& type) {
-  // TODO(anyone): Implementation goes here
+  Assert(row_count() == 0, "The table already contains rows, column scheme can not be altered anymore.");
+  _column_definitions.emplace_back(name, type);
+}
+
+void Table::copy_column_definition(const std::shared_ptr<const Table>& other_table, const ColumnID column_id) {
+  _column_definitions.push_back(other_table->_column_definitions.at(column_id));
 }
 
 void Table::add_column(const std::string& name, const std::string& type) {
-  Assert(row_count() == 0, "The table already contains rows, column scheme can not be altered anymore.");
-  _columns.emplace_back(name, type);
+  add_column_definition(name, type);
   _append_column_to_chunks(type);
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
   if (_chunks.back()->size() == _target_chunk_size) {
-    _append_new_chunk();
+    create_new_chunk();
   }
   _chunks.back()->append(values);
 }
 
 void Table::create_new_chunk() {
-  // TODO(anyone): Implementation goes here
+  auto new_chunk = std::make_shared<Chunk>();
+
+  for (const auto& column : _column_definitions) {
+    // append existing columns as segments to new chunk
+    new_chunk->add_segment(_create_value_segment_for_type(column.type));
+  }
+  _chunks.emplace_back(new_chunk);
 }
 
 // TODO(max): write test
```

Comment:
Not sprint 3, but 
```
Assert(_chunks.back()->size() != _target_chunk_size,
           "Cannot emplace chunk because current last chunk is not full.");
    _chunks.emplace_back(chunk);
```

is untested

Reply:
We deleted the assertion, but will write a test for emplace_back